splashsurf/src/io/ply_format.rs

```diff
@@ -49,3 +53,139 @@ pub fn particles_from_ply<R: Real, P: AsRef<Path>>(
+        .into_iter()
+        .map(|e| {
+            let vertex = (
+                e.get("x").unwrap(),
+                e.get("y").unwrap(),
+                e.get("z").unwrap(),
+                e.get("nx").unwrap(),
+                e.get("ny").unwrap(),
+                e.get("nz").unwrap(),
+            );
+
+            let v = match vertex {
+                (
+                    Property::Float(x),
+                    Property::Float(y),
+                    Property::Float(z),
+                    Property::Float(nx),
+                    Property::Float(ny),
+                    Property::Float(nz),
+                ) => (
+                    Vector3::new(
+                        R::from_f32(*x).unwrap(),
+                        R::from_f32(*y).unwrap(),
+                        R::from_f32(*z).unwrap(),
+                    ),
+                    Vector3::new(
+                        R::from_f32(*nx).unwrap(),
+                        R::from_f32(*ny).unwrap(),
+                        R::from_f32(*nz).unwrap(),
+                    ),
+                ),
+                _ => {
+                    return Err(anyhow!(
+                        "Vertex properties have wrong PLY data type (expected float)"
+                    ))
+                }
+            };
+
+            Ok(v)
+        })
+        .map(|vn| vn.unwrap())
+        .collect();
+
+    let vertices: Vec<Vector3<_>> = vertices_normals.iter().map(|vn| vn.0.clone()).collect();
+    let normals: Vec<Vector3<_>> = vertices_normals.iter().map(|vn| vn.1.clone()).collect();
+
+    let faces = ply
+        .payload
+        .get("face")
+        .ok_or(anyhow!("PLY file is missing a 'face' element"))?;
+
+    let triangles = faces
+        .into_iter()
+        .map(|e| {
+            // This is as per what blender creates for a
+            let indices = e.get("vertex_indices");
+            if let Some(indices) = indices {
+                if let Property::ListUInt(indices) = indices {
+                    if indices.len() == 3 {
+                        return Ok([
+                            indices[0] as usize,
+                            indices[1] as usize,
+                            indices[2] as usize,
+                        ]);
+                    } else {
+                        return Err(anyhow!(
+                            "Invalid number of vertex indices per cell: {}",
+                            indices.len()
+                        ));
+                    }
+                } else {
+                    return Err(anyhow!(
+                        "Index properties have wrong PLY data type (expected uint)"
+                    ));
+                }
+            } else {
+                return Err(anyhow!(
+                    "Vertex properties have wrong PLY data type (expected uint)"
+                ));
+            }
+        })
+        .map(|e| e.unwrap())
+        .collect();
```

Comment:
Btw. this can be replaced with a simple
```rust
.collect::<Result<Vec<_>, _>>()?;
```
You can collect `Result<T, E>`s into a `Result<Vec<T>, E>` which stops on first error.